x-pack/plugins/apm/public/hooks/useComponentId.tsx

```diff
@@ -4,8 +4,12 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-export const STATUS = {
-  FAILURE: 'FAILURE',
-  SUCCESS: 'SUCCESS',
-  LOADING: 'LOADING'
-};
+import { useRef } from 'react';
+
+let uniqueId = 0;
+const getUniqueId = () => uniqueId++;
+
+export function useComponentId() {
+  const idRef = useRef(getUniqueId());
```

Comment:
it seems that getUniqueId would still be called even between renders on the same component instance. I think that means the id would be incremented each render rather than each component? this shouldn't be an issue since it will still be unique, and the initial value will stay the same. But i'm not sure if this is intended.

Reply:
Yes, true. The id is stable within a component but is incremented between renders, so it will become quite high.
Do you see a good solution for this?

Reply:
If it's a concern you could set `idRef.current = getUniqueId()` within an `if (!idRef.current)` statement, which will only call `getUniqueId` once per instance rather than each render, But if it's not an issue, just leave as is.  I just wanted to make u you were aware.

Reply:
Good point. It doesn't appear to be an issue so I'll leave it as-is.